FrameworkSystem/DB/AuthDB.py

```diff
@@ -0,0 +1,336 @@
+""" Auth class is a front-end to the Auth Database
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import re
+import json
+import pprint
+import random
+import string
+
+from ast import literal_eval
+from datetime import datetime
+
+from DIRAC import gConfig, S_OK, S_ERROR, gLogger
+from DIRAC.Core.Base.DB import DB
+
+__RCSID__ = "$Id$"
+
+# Session statuses
+
+# successfully authed and ready to use
+SESSION_READY = "authed"
+# crashed
+SESSION_FAILED = "failed"
+# authed, paused for waiting when second flow will end
+SESSION_REDIRECT = "redirect"
+# just created
+SESSION_PREPARED = "prepared"
+# first request to get auth link
+SESSION_PROGRESS = "in progress"
+# finishing
+SESSION_FINISHING = "finishing"
+
+
+class AuthDB(DB):
```

Comment:
There's no integration test of this class...?

Reply:
in progress